cmd/tsbs_generate_queries/databases/timescaledb/devops_test.go

```diff
@@ -152,3 +154,333 @@ func TestDevopsGetSelectClausesAggMetrics(t *testing.T) {
+			expectedHumanDesc:  "TimescaleDB mean of 1 metrics, all hosts, random 12h0m0s by 1h: 1970-01-01T00:37:12Z",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: `
+        WITH cpu_avg AS (
+          SELECT time_bucket('3600 seconds', time) as hour, tags_id,
+          avg(usage_user) as mean_usage_user
+          FROM cpu
+          WHERE time >= '1970-01-01 00:37:12.342805 +0000' AND time < '1970-01-01 12:37:12.342805 +0000'
+          GROUP BY hour, tags_id
+        )
+        SELECT hour, tags->>'hostname', mean_usage_user
+        FROM cpu_avg
+        JOIN tags ON cpu_avg.tags_id = tags.id
+        ORDER BY hour, tags->>'hostname'`,
+		},
+	}
+
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(13 * time.Hour)
+
+	numMetrics := 1
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			d := NewDevops(s, e, 10)
+			d.UseJSON = c.useJSON
+			d.UseTags = c.useTags
+
+			q := d.GenerateEmptyQuery()
+			d.GroupByTimeAndPrimaryTag(q, numMetrics)
+
+			verifyQuery(t, q, c.expectedHumanLabel, c.expectedHumanDesc, c.expectedHypertable, c.expectedSQLQuery)
+		})
+	}
+}
+
+func TestMaxAllCPU(t *testing.T) {
+	expectedHumanLabel := "TimescaleDB max of all CPU metrics, random    1 hosts, random 8h0m0s by 1h"
+	expectedHumanDesc := "TimescaleDB max of all CPU metrics, random    1 hosts, random 8h0m0s by 1h: 1970-01-01T02:16:22Z"
+	expectedHypertable := "cpu"
+	expectedSQLQuery := `SELECT time_bucket('3600 seconds', time) AS hour,
+        max(usage_user) as max_usage_user, max(usage_system) as max_usage_system, max(usage_idle) as max_usage_idle, ` +
+		"max(usage_nice) as max_usage_nice, max(usage_iowait) as max_usage_iowait, max(usage_irq) as max_usage_irq, " +
+		"max(usage_softirq) as max_usage_softirq, max(usage_steal) as max_usage_steal, max(usage_guest) as max_usage_guest, " +
+		`max(usage_guest_nice) as max_usage_guest_nice
+        FROM cpu
+        WHERE (hostname = 'host_9') AND time >= '1970-01-01 02:16:22.646325 +0000' AND time < '1970-01-01 10:16:22.646325 +0000'
+        GROUP BY hour ORDER BY hour`
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(13 * time.Hour)
+
+	d := NewDevops(s, e, 10)
+
+	q := d.GenerateEmptyQuery()
+	d.MaxAllCPU(q, 1)
+	verifyQuery(t, q, expectedHumanLabel, expectedHumanDesc, expectedHypertable, expectedSQLQuery)
+}
+
+func TestLastPointPerHost(t *testing.T) {
+	cases := []struct {
+		desc               string
+		useJSON            bool
+		useTags            bool
+		expectedHumanLabel string
+		expectedHumanDesc  string
+		expectedHypertable string
+		expectedSQLQuery   string
+	}{
+		{
+			desc:               "no JSON or tags",
+			expectedHumanLabel: "TimescaleDB last row per host",
+			expectedHumanDesc:  "TimescaleDB last row per host",
+			expectedHypertable: "cpu",
+			expectedSQLQuery:   "SELECT DISTINCT ON (hostname) * FROM cpu ORDER BY hostname, time DESC",
```

Comment:
And with queries that are not constant but have specific parts that are filled in, I suggest that we keep the fixed parts as templates in constant, and then format them with the generated values. 

This way the tests will be a lot shorter and all the queries will be in one place in one file for a quick review

Reply:
My reasoning was keeping the same style of test for all the query generation functions. Thats why I'm testing a fixed string (which could change to a dynamic one in the future).

As for making the tests shorter by reusing the templates, I guess you would have to basically rewrite all the logic in the function by reusing the same functions used to fill out the template and the test would be mirroring the function in test. I'm not fond of that approach since you are just writing the same logic twice and checking that it generates the same output. I decided to go this way so we can at least see the whole query visually in the test to check for sanity.

I, also, wanted to make a point of not changing the current code until we have some test coverage. After that we can introduce some refactoring at which point we can refactor the tests if there is a need for it.

Reply:
Yeah I do like the approach for now of having the whole query, even if it is a bit cluttered. I think @blagojts 's suggestions for that function are good ones for a follow up PR (since as @antekresic mentioned, we will then have tests to cover it in case the refactor goes wrong). Those changes being making the strings `const`s and not use `Sprintf` for no reason.